src/Playground.res

```diff
@@ -1100,8 +1100,93 @@ module Settings = {
     </div>
   }
 }
+module Transpiler = {
+  @module("./ffi/transpile-to-eval") external transpile: string => string = "default"
+
+  let srcdoc = `
+        <!DOCTYPE html>
+          <html lang="en">
+            <head>
+              <meta charset="UTF-8" />
+              <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+              <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+              <title>Document</title>
+            </head>
+
+            <body>
+              <div id="root"></div>
+                <script
+                  src="https://unpkg.com/react@17/umd/react.production.min.js"
+                  crossorigin
+                ></script>
+                <script
+                  src="https://unpkg.com/react-dom@17/umd/react-dom.production.min.js"
+                  crossorigin
+                ></script>
+                <script
+                  src="https://bundleplayground.s3.sa-east-1.amazonaws.com/bundle.js"
+                  crossorigin
+                ></script>
+                <script>
+                  window.addEventListener("message", (event) => {
+                    const mainWindow = event.source;
+                    let result = "all good";
+                    try {
+                      eval(event.data);
+                  } catch (err) {
+                    console.log(err);
+                    result = "eval() threw an exception.";
+                  }
+                  mainWindow.postMessage(result, event.origin);
+                 });
+              </script>
+  </body>
+</html>
+      `
+  type document
+  type contentWindow = {
+    @uncurry
+    postMessage: (. string, string) => unit,
+  }
+  type element = {contentWindow: option<contentWindow>}
+  @send external getElementById: (document, string) => Js.nullable<element> = "getElementById"
+  @val external doc: document = "document"
+
+  let runCode = (~code: string) => {
+    let iframeWin = Js.toOption(getElementById(doc, "iframe-eval"))
+    switch iframeWin {
+    | Some(element) =>
+      switch element.contentWindow {
+      | Some(win) => {
+          let codeToRun = `(function () {
+          ${transpile(code)}
+          const root = document.getElementById("root");
+          ReactDOM.render(App.make(), root);
+        })();`
+          win.postMessage(. codeToRun, "*")
+        }
+      | None => ()
+      }
+    | None => ()
+    }
+  }
+}
 
 module ControlPanel = {
+  let codeFromResult = (result: FinalResult.t): string => {
+    open Api
+    switch result {
+    | FinalResult.Comp(comp) =>
+      switch comp {
+      | CompilationResult.Success({js_code}) => js_code
+      | UnexpectedError(_)
+      | Unknown(_, _)
+      | Fail(_) => "/* No JS code generated */"
+      }
+    | Nothing
+    | Conv(_) => "/* No JS code generated */"
+    }
+  }
```

Comment:
Moved this from the [Output Panel](https://github.com/rescript-association/rescript-lang.org/pull/420/files#diff-586dd1eeb35d17d13f2757b5807aa6317b2f555e064b5d018fe95c3ba3f2651eR1343) in order to re-use it here.